src/routeStep/controller.ts

```diff
@@ -31,6 +32,46 @@ function putRouteStep(req: express.Request, res: express.Response) {
   );
 }
 
+/**
+ * Update a RouteStep with the given category and name.
+ */
+function patchRouteStep(req: express.Request, res: express.Response) {
+  ExpressHandler(
+    async () => {
+      const { routeStepId, category, name } = req.body;
+
+      if (!routeStepId)
+        return HTTPError('Route Step ID must be specified.', 400);
+
+      const routeStep = await RouteStepModel.findById(routeStepId);
+
+      if (!routeStep) return HTTPError('Step does not exist.', 404);
+
+      let updateDict = {};
+
+      if (category) {
+        updateDict = { ...updateDict, category };
```

Comment:
Rewriting this as `Object.assign(updateDict, category)`; same below. for performance reasons.

This lets us declare `const updateDict = {}` which will also be more performant.